snap/db.go

```diff
@@ -32,6 +32,9 @@ func (s *Snapshotter) SaveDBFrom(r io.Reader, id uint64) error {
 		return err
 	}
 	_, err = io.Copy(f, r)
+	if err == nil {
+		err = f.Sync()
+	}
 	f.Close()
```

Comment:
We should also handle the error returned by f.Close(). But it is another issue. I will send another pull request to fix it.